src/realm/query.cpp

```diff
@@ -1469,20 +1469,22 @@ void Query::find_all(ConstTableView& ret, size_t begin, size_t end, size_t limit
 
     init();
 
+    bool has_cond = has_conditions();
+
     if (m_view) {
         if (end == size_t(-1))
             end = m_view->size();
         for (size_t t = begin; t < end && ret.size() < limit; t++) {
             const Obj obj = m_view->try_get_object(t);
-            if (obj && eval_object(obj)) {
+            if (!has_cond || (obj && eval_object(obj))) {
```

Comment:
Ok, good. So was I wrong to add [this check on obj](https://github.com/realm/realm-core/pull/4594/files#diff-f635cbd3382d46174a9185c4dc677d0ef78267a958becde735479aaac4a2ff91R1458)? Maybe it would read better to move that check on obj down into eval_object?

Also I think the same fix will need to be added to Query::find() [above](https://github.com/realm/realm-core/pull/4636/files#diff-f635cbd3382d46174a9185c4dc677d0ef78267a958becde735479aaac4a2ff91R1436).

Reply:
Yes, better to move the check into `eval_object`.